pilot/pkg/proxy/envoy/v2/eds.go

```diff
@@ -491,17 +491,19 @@ func (s *DiscoveryServer) WorkloadUpdate(id string, labels map[string]string, an
 // on each step: instead the conversion happens once, when an endpoint is first discovered.
 func (s *DiscoveryServer) EDSUpdate(shard, serviceName string,
```

Comment:
Given `edsUpdate` never returns error, so we can change `EDSUpdate` to not return error as well. Then the kube/controller[1] code can be simplified as well.

@costinm Let me know if this makes sense.

[1] https://github.com/istio/istio/blob/master/pilot/pkg/serviceregistry/kube/controller.go#L792